Cubical/ZCohomology/GroupStructure.agda

```diff
@@ -211,6 +211,28 @@ assocₖ (suc zero) =
                     ≡ (cong (_+ₖ ∣ base ∣) (sym (rUnitₖ 1 ∣ x ∣)))
   helper = toPropElim (λ _ → isOfHLevelTrunc 3 _ _ _ _)
                       (sym (lUnit refl))
+assocₖ (suc (suc n)) =
+  trElim3 (λ _ _ _ → isOfHLevelPath (4 + n) (isOfHLevelTrunc (4 + n)) _ _)
+          (wedgeConSn-×3 _
+            (λ x z i → preAdd n .snd .snd x (~ i) +ₖ ∣ z ∣)
+            (λ x y → cong (∣ x ∣ +ₖ_) (rUnitₖ (2 + n) ∣ y ∣) ∙ sym (rUnitₖ (2 + n) (∣ x ∣ +ₖ ∣ y ∣)))
+            (lUnit (sym (rUnitₖ (2 + n) (∣ north ∣ +ₖ ∣ north ∣)))))
+  where
+  wedgeConSn-×3 : (n : ℕ)
+          → (f : (x z : S₊ (2 + n))→ ∣ x ∣ +ₖ ((0ₖ _) +ₖ ∣ z ∣) ≡ (∣ x ∣ +ₖ (0ₖ _)) +ₖ ∣ z ∣)
+          → (g : (x y : S₊ (2 + n)) → ∣ x ∣ +ₖ (∣ y ∣ +ₖ 0ₖ _) ≡ (∣ x ∣ +ₖ ∣ y ∣) +ₖ 0ₖ _)
+          → (f (ptSn _) (ptSn _) ≡ g (ptSn _) (ptSn _))
+          → (x y z : S₊ (2 + n)) → ∣ x ∣ +ₖ (∣ y ∣ +ₖ ∣ z ∣) ≡ (∣ x ∣ +ₖ ∣ y ∣) +ₖ ∣ z ∣
+  wedgeConSn-×3 n f g d x =
+    wedgeConSn _ _ (λ _ _ → isOfHLevelPath ((2 + n) + (2 + n)) (wedgeConHLev n) _ _)
+               (f x)
+               (g x)
+               (sphereElim _ {A = λ x → g x (ptSn (suc (suc n))) ≡ f x (ptSn (suc (suc n))) }
+                             (λ _ → isOfHLevelTrunc (4 + n) _ _ _ _)
+                             (sym d) x) .fst
+{-
+This was the original proof for the case n ≥ 2:
+For some reason it doesn't check in reasonable time anymore:
```

Comment:
Hmm, it's truly mysterious to me why this would be so much slower... @Saizan Any clue?

Reply:
It seems like the type of `helper` has a massive normal form, so if equality checking ends up expanding that type we will get such a slowdown.

The new code avoids this by working with the local variables `f` and `g` which are not going to reduce, and the call to `wedgeConSn-×3` does not seem to involve types as complicated as the one for `helper`.

Reply:
I see, it's scary that the type of helper is so huge...